### DISCUSSION TITLE
GitHub repo for React course not showing a language

### DISCUSSION THREAD

**Discussion by milan-mal** [github-linguist/linguist#6239] [Classification] [answered]
repo: https://github.com/milan-mal/fullstack-open-22

Hi,
I'm going through a MOOC full-stack React course. My code is in the repo above.
I don't see a language defined for the repo. I was able to see it initially.
I read through the linguist documentation and couldn't find why JavaScript is not recognized.
Thank you

**Comment by lildude** [comment, marked as answer]
It's because you have more than 100k files in your repo:

```
$ ls-files | wc -l
  147545
$
```

Linguist will not analyse a repo with more [than 100k files](https://github.com/github/linguist/blob/c1c34e5260797b4d598f5ec76f19723bfc5a1894/lib/linguist/repository.rb#L132) for performance reasons.

Remove all the `node_modules` directories and you'll save yourself a lot of space, time, effort and get back your language stats. 

ProTip: add this directory to a `.gitignore` file in your repo to prevent it accidentally being included in future commits.
